h/models/user.py

```diff
@@ -103,6 +103,23 @@ def __eq__(self, other):
                                   val['domain'])
         return self.__clause_element__() == other
 
+    def in_(self, userids):
```

Comment:
So this is a helper used to generate `<user ID expr> IN (<other user ID 1 expr>, <other user ID 2 expr>, ...)` SQL expressions?

Reply:
Yes, it's a custom implementation of the default SQLAlchemy `sqlalchemy.orm.Query.in_` function. We don't store the userid in the user table, which is why we need to implement these custom methods that translate a userid to a username and authority. So far we only needed the `a = b` comparator function (`__eq__`), but now we want to do a `IN` query which means we have to implement this ourselves.